text/0000-buildpacks-config.md

```diff
@@ -0,0 +1,314 @@
+
+## File
+
+`buildpacks.<ext>` where `<ext>` corrolates to a supported format.
+
+By using the term "buildpacks" in the file name, it becomes immediately apparent that this project has custom configuration for Cloud Native Buildpacks.
+
+### Supported Formats
+
+By supporting multiple file formats, we lower the barrier to entry and allow for the flexibility desired by some app developers.
+
+Proposed supported extensions (in order of precedence):
+
+ - `toml` => TOML
+ - `yaml`, `yml` => YAML
+
+#### TOML
+
+```toml
+schema="0.1"
+
+[images]
+names=[
+    "cnbs/sample-app",
+    "cnbs/sample-app:v1",
+    "grc.io/cnbs/sample-app:v1"
+]
+
+[images.builder]
+name="cnbs/sample:builder"
+trusted=true
+
+[images.previous]
+name="cnbs/sample-app"
+
+[images.run]
+name = "cnbs/sample-stack-run:bionic"
+mirrors = [
+	"grc.io/cnbs/sample-stack-run:bionic",
+]
+
+[[env]] # build-time env vars
+name="ENV_1"
+value="ENV_2"
+operation="append" # default=override
+
+[[buildpacks]]
+id = "samples/hello-world"
+version = "0.0.1"
+
+[[buildpacks]]
+id = "samples/java-maven"
+version = "0.0.1"
+
+###
+# Contents Configuration
+###
+
+[source]
+workspace = "/workspace"
+# exclude = [] # mutially exclusive with 'include'
+include = [
+    "cmd/",
+    "go.mod",
+    "go.sum",
+    "*.go"
+]
+
+##
+# Caching
+# See https://github.com/buildpacks/rfcs/blob/main/text/0091-pack-cache-options.md
+##
+
+[cache]
+format="image"
+name="cnbs/sample-app-cache:build"
+
+## Alternatives:
+#
+# [cache]
+# format="volume"
+# name=""
+#
+# [cache]
+# format="bind"
+# source="./cache"
+
+###
+# Process Specific Configuration
+###
+
+[process]
+default = "web"
+
+[[process.web.env]] # runtime env vars
+name="PATH"
+value="ENV_2"
+operation="append" # default=override
+```
+
+#### YAML
+
+```yaml
+schema: "0.1"
+images:
+  names:
+    - cnbs/sample-app
+    - 'cnbs/sample-app:v1'
+    - 'grc.io/cnbs/sample-app:v1'
+  builder:
+    name: 'cnbs/sample:builder'
+    trusted: true
+  previous:
+    name: cnbs/sample-app
+  run:
+    name: 'cnbs/sample-stack-run:bionic'
+    mirrors:
+      - 'grc.io/cnbs/sample-stack-run:bionic'
+env:
+  - name: ENV_1
+    value: ENV_2
+    operation: append
+buildpacks:
+  - id: samples/hello-world
+    version: 0.0.1
+  - id: samples/java-maven
+    version: 0.0.1
+source:
+  workspace: /workspace
+  include:
+    - cmd/
+    - go.mod
+    - go.sum
+    - '*.go'
+cache:
+  format: image
+  name: 'cnbs/sample-app-cache:build'
+process:
+  default: web
+  web:
+    env:
+      - name: PATH
+        value: ENV_2
+        operation: append
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+### Configuration Properties
+
+To prevent unnecessary complexity for end-users (app developers), the `buildpacks.<ext>` file would only support properties associated with Cloud Native Buildpacks. See full list of properties in the [file](#file) section.
+
+### Cross-Platform Support
```

Comment:
This doesn't appear to improve on the status quo (`project.toml`). The `buildpacks.*` would. need to be converted into another format to become cross-platform supportable.

Reply:
It does in that Platforms become required to support `buildpacks.*`. That said, we could do the same to `project.toml`. This RFC tries to paint a _new_ picture on a CNB specific config file that becomes required for platforms to support.

Reply:
what are the advantages of requiring this file over `project.toml`?